packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -597,7 +597,11 @@ export class CfnInclude extends core.CfnElement {
       l1Instance = this.createNestedStack(logicalId, cfnParser);
     } else {
       const l1ClassFqn = cfn_type_to_l1_mapping.lookup(resourceAttributes.Type);
-      if (l1ClassFqn) {
+      // The AWS::CloudFormation::CustomResource type corresponds to the CfnCustomResource class.
+      // Unfortunately, it's quite useless; it only has a single property, ServiceToken.
```

Comment:
So one option is what you've done here -- mapping a custom resource to a CfnResource. Would the other be to fix the "useless" CfnCustomResource class? Could we add extensions to that class via the cfnspec additions? If so, it seems like that would solve this issue for more than just this one use case.

Reply:
One advantage of this escape hatch to `CfnResource` is that `AWS::CloudFormation::CustomResource` is handled the same as `Custom::MyCoolResource` instead of one being `CfnResource` and the other being `CfnCustomResource`

Reply:
Unfortunately, `CfnCustomResource` is auto-generated, so there's not much we can do there 😕. There's simply not a concept of "accept an arbitrary collection of properties" in the CloudFormation resource specification, and so it can't really be translated to our generated L1s.

We can try to hack something on top of the resource specification (we've done that before, for example in #10316), but I'm not sure it's worth it in this case - like @DV8FromTheWorld said, it's actually more consistent in some sense to always return custom resources, regardless of the resource type they use, as instances of the `CfnResource` class.

Reply:
You know, maybe I spoke too soon. I see that the resource specification for `AWS::CloudFormation::Resource` contains an `"AdditionalProperties": true` field:

```json
    "AWS::CloudFormation::CustomResource": {
      "AdditionalProperties": true,
      "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cfn-customresource.html",
      "Properties": {
        "ServiceToken": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cfn-customresource.html#cfn-customresource-servicetoken",
          "PrimitiveType": "String",
          "Required": true,
          "UpdateType": "Immutable"
        }
      }
    },
```

Not sure if it's possible to express that easily in TypeScript, but I can investigate.

Let me know what you think @njlynch.

Reply:
Thanks for the deeper investigation. I think the context I was missing was that we return the other custom resource types as `CfnResource` as well. For the sake of consistency, I think we can stick with this approach.